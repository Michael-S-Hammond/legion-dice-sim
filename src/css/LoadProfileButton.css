.load-profile-button {
  display: flex;
  width: 30px;
  height: 30px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px;
  background-image: url("../img/person-plus-fill.svg"); }
