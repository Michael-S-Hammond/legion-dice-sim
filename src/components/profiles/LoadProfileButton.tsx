import '../../css/LoadProfileButton.css';

import React from 'react';

type LoadProfileButtonProps = {
    dialogId: string,
    tooltip: string,
};

class LoadProfileButton extends React.Component<LoadProfileButtonProps> {
    constructor(props : LoadProfileButtonProps) {
        super(props);
    }

    render() : JSX.Element {
        return (
            <button
                className="btn btn-light border rounded-lg load-profile-button mx-2 my-auto"
                title={this.props.tooltip}
                data-toggle="modal"
                data-target={"#" + this.props.dialogId}>
            </button>
        );
    }
}

export default LoadProfileButton;
