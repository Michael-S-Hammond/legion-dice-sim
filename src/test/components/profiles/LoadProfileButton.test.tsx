import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import LoadProfileButton from '../../../components/profiles/LoadProfileButton';

describe('FactionButtonGroup', () => {
    it('matches the snapshot', () => {
        const onFactionChange = jest.fn();
        const button = shallow(<LoadProfileButton
                dialogId='fakeDialog'
                tooltip='My tooltip'
            ></LoadProfileButton>);
        const snapshot = shallowToJson(button);
        expect(snapshot).toMatchSnapshot();
    });
});
