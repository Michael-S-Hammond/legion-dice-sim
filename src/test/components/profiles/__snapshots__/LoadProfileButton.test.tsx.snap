// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactionButtonGroup matches the snapshot 1`] = `
<button
  className="btn btn-light border rounded-lg load-profile-button mx-2 my-auto"
  data-target="#fakeDialog"
  data-toggle="modal"
  title="My tooltip"
/>
`;
